Tests/Unit/Model/EpsPaymentMethodTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use Wirecard\Oxid\Core\PaymentMethodHelper;
+use Wirecard\Oxid\Model\EpsPaymentMethod;
+use Wirecard\Oxid\Model\SepaCreditTransferPaymentMethod;
+
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\Config\PaymentMethodConfig;
+use Wirecard\PaymentSdk\Transaction\EpsTransaction;
+
+use OxidEsales\Eshop\Application\Model\Payment;
+
+class EpsPaymentMethodTest extends OxidEsales\TestingLibrary\UnitTestCase
+{
+    /**
+     * @var EpsPaymentMethod
+     */
+    private $_oPaymentMethod;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->_oPaymentMethod = new EpsPaymentMethod();
+    }
+
+    public function testGetConfig()
+    {
+        $oPayment = PaymentMethodHelper::getPaymentById(EpsPaymentMethod::getName(true));
+
+        $oConfig = $this->_oPaymentMethod->getConfig();
+
+        $this->assertInstanceOf(Config::class, $oConfig);
+        $this->assertInstanceOf(PaymentMethodConfig::class, $oConfig->get(EpsPaymentMethod::getName()));
+    }
+
+    public function testGetTransaction()
+    {
+        $oTransaction = $this->_oPaymentMethod->getTransaction();
+        $this->assertInstanceOf(EpsTransaction::class, $oTransaction);
+    }
+
+    /**
+     * @dataProvider getNameProvider
+     */
+    public function testGetName($bForOxid, $sExpected)
+    {
+        $sName = EpsPaymentMethod::getName($bForOxid);
+        $this->assertEquals($sExpected, $sName);
+    }
+
+    public function getNameProvider()
+    {
+        return [
+            'for oxid' => [true, 'wdeps'],
+            'not for oxid' => [false, 'eps'],
+        ];
+    }
+
+    public function testGetConfigFields()
+    {
+        $aFields = $this->_oPaymentMethod->getConfigFields();
+        $this->assertEquals([
+            'apiUrl',
+            'httpUser',
+            'httpPassword',
+            'testCredentials',
+            'maid',
+            'secret',
+            'descriptor',
+            'additionalInfo',
+            'deleteCanceledOrder',
+            'deleteFailedOrder',
+        ], array_keys($aFields));
+    }
+
+    public function testGetPublicFieldNames()
+    {
+        $aPublicFieldNames = $this->_oPaymentMethod->getPublicFieldNames();
+        $aExpected = [
+            "apiUrl",
+            "maid",
+            "descriptor",
+            "additionalInfo",
+            "deleteCanceledOrder",
+            "deleteFailedOrder",
+        ];
+        $this->assertEquals($aExpected, $aPublicFieldNames, '', 0.0, 1, true);
```

Comment:
What do the last 2 arguments represent here? I couldn't find such a case in the docs.

Reply:
I also couldn't find it in the documentation now. We've used the same test case for other payment methods (Credit Card, PayPal, SEPA Credit Transfer).

Reply:
@ps-cgrach @ps-mstifter Any input here? Shall we remove those redundant arguments?

Reply:
Ok, never mind, I checked the source code now:
```php
function assertEquals($expected, $actual, $message = '', $delta = 0.0, $maxDepth = 10, $canonicalize = false, $ignoreCase = false)
{ /* ... */ }
```
If we need those, keep it like that.

Reply:
So the important part is the last parameter, the `canonicalize` parameter. It tells the `assert` that we don't care about the order of the items in the array. I'll create a "helper" assert to have it more obvious what it does!